src/modelframe.jl

```diff
@@ -172,19 +172,40 @@ function setcontrasts!(mf::ModelFrame, new_contrasts::Dict)
 end
 setcontrasts!(mf::ModelFrame; kwargs...) = setcontrasts!(mf, Dict(kwargs))
 
+convertinsupport(v::AbstractVector, d::Distribution) = Vector{partype(d)}(v)
+
+function convertinsupport(v::Union{AbstractCategoricalVector, AbstractNullableCategoricalVector},
+    d::Bernoulli)
+    vdrp = droplevels!(copy(v))
```

Comment:
Rather than allocating a copy of `v`, would be more efficient to do `levs = intersect(levels(v), unique(v))`. (I should add a `drop` argument to `levels` which would do exactly that.)